torchgeo/datamodules/geo.py

```diff
@@ -220,6 +248,41 @@ def setup(self, stage: str) -> None:
                 self.test_dataset, self.patch_size, self.patch_size
             )
 
+    def _dataloader_factory(self, split: str) -> DataLoader[dict[str, Tensor]]:
+        """Implement one or more PyTorch DataLoaders.
+
+        Args:
+            split: Either 'train', 'val', 'test', or 'predict'.
```

Comment:
In setup() there is no case for `predict` if I am not misunderstanding something.

Reply:
There is, we just don't set it by default because most datasets don't have a `predict` split.